### PR TITLE
feat(Backend-zsilentcoders): implementing user authentication

### PR DESCRIPTION
**Pull request by yonass08**
This pull request updates the user authentication process in the backend project by implementing JSON Web Token (JWT) based authentication.

In this update, we have added a new authentication middleware that verifies the JWT token and grants access to the authorized user. The JWT token is generated at the time of user login and is then used to authenticate the user for subsequent requests. The token contains user information and is signed using a secret key to ensure its authenticity.

We use a separate Identity database for the login service which is provided by Identity framework. We used it in conjuction with MailKit to send emails for password reset and confirming emails. And we used google smtp server with our personal account to send the emails.

We also implemented a separate user model so that during integration phase it can easily be integrated with other models, since it is on the same database. And we linked this user to the identity user by including a field called AccountId which refers to the identity user on the identity database.

we will remove the delete and create actions on the user controller and the delete action in the authcontroller, as they are only there for testing purposes and shouldn't be on the production version.

### CODE REVIEW

**Comment by danielteferadeti** [comment]
Very nice description of your task! Best one I have seen sofar...thank you for that!

File: Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Application/Features/Users/CQRS/Handlers/Get_UserListQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        public Get_UserListQueryHandler(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task<List<_UserDto>> Handle(Get_UserListQuery request, CancellationToken cancellationToken)
+        {
+            var users = await _unitOfWork._UserRepository.GetAll();
+            Console.Write(users);
```
Comment (danielteferadeti) [inline Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Application/Features/Users/CQRS/Handlers/Get_UserListQueryHandler.cs:28]: This should be removed.

File: Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Identity/Configurations/RoleConfiguration.cs
Comment (danielteferadeti) [inline Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Identity/Configurations/RoleConfiguration.cs:1]: This section look good

File: Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Application/Features/Users/CQRS/Handlers/Get_UserListQueryHandler.cs (hunk 1 above)
Comment (yonass08) [reply Backend/starter_project_zsilentcoders/BlogApp3/BlogApp.Application/Features/Users/CQRS/Handlers/Get_UserListQueryHandler.cs]: We have removed it

**Review by danielteferadeti — APPROVED** [review]
Looks good!

**Pull request merged**